nautobot/extras/forms/mixins.py

```diff
@@ -687,6 +691,43 @@ def _append_relationships_side(self, relationships, initial_side):
             self.relationships.append(field_name)
 
 
+class RoleModelBulkEditFormMixin(forms.Form):
+    """Mixin to add non-required `role` choice field to forms."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["role"] = DynamicModelChoiceField(
+            required=False,
+            queryset=Role.objects.all(),
+            query_params={"content_types": self.model._meta.label_lower},
+        )
+        self.order_fields(self.field_order)  # Reorder fields again
+
+
+class RoleModelFilterFormMixin(forms.Form):
+    """
+    Mixin to add non-required `role` multiple-choice field to filter forms.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["role"] = DynamicModelMultipleChoiceField(
+            required=False,
+            queryset=Role.objects.all(),
+            query_params={"content_types": self.model._meta.label_lower},
+            to_field_name="name",  # TODO(timizuo): Might be slug but i dont think so
```

Comment:
At the very least it appears to be inconsistent with `StatusModelFilterFormMixin` -- intentional?